frontend/lib/ui/artefact_dialog/test_execution_review.dart

```diff
@@ -40,6 +37,7 @@ class TestExecutionReviewButton extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
+    final artefactId = AppRoutes.artefactIdFromContext(context);
```

Comment:
given that this widget doesn't use `artefactId` perhaps `TestExecutionPopOver` should be the one to get it from context?

Reply:
Actually, I tried this initially and it didn't work, I think it is because the Popover is a separate layer in the render. After some searching, I found out about this [open issue in Flutter](https://github.com/flutter/flutter/issues/131154) which seems to be going resolved in a [separate issue](https://github.com/flutter/flutter/issues/129833) but doesn't seem to be closed yet.